File: samples/ObjC Manual Sample/watchkitapp Extension/RegistrationController.m
/* 
 * Licensed Materials - Property of IBM 
 * 
 * 5725E28, 5725I03 
 * 
 * © Copyright IBM Corp. 2017, 2018
 * US Government Users Restricted Rights - Use, duplication or disclosure restricted by GSA ADP Schedule Contract with IBM Corp. 
 */

#import "RegistrationController.h"
#import <IBMMobilePushWatch/IBMMobilePushWatch.h>

@interface  RegistrationController()
@property id observer;
@end

@implementation RegistrationController

- (void)willDisappear
{
    [super willDisappear];
    [NSNotificationCenter.defaultCenter removeObserver:self.observer];
}

- (void)willActivate
{
    [super willActivate];
    [self updateRegistrationLabels];
    self.observer = [NSNotificationCenter.defaultCenter addObserverForName:MCERegisteredNotification object:nil queue:NSOperationQueue.mainQueue usingBlock:^(NSNotification*notification){
        [self updateRegistrationLabels];
    }];
}

-(void)updateRegistrationLabels
{
    if(MCERegistrationDetails.sharedInstance.mceRegistered)
    {
        [self.userIdLabel setText: MCERegistrationDetails.sharedInstance.userId];
        [self.channelIdLabel setText: MCERegistrationDetails.sharedInstance.channelId];
        [self.appKeyLabel setText: MCERegistrationDetails.sharedInstance.appKey];
    }
}

@end

File: samples/Swift Combined Inbox Sample/SampleWatch Extension/RegistrationController.swift
/* 
 * Licensed Materials - Property of IBM 
 * 
 * 5725E28, 5725I03 
 * 
 * © Copyright IBM Corp. 2017, 2018
 * US Government Users Restricted Rights - Use, duplication or disclosure restricted by GSA ADP Schedule Contract with IBM Corp. 
 */

import Foundation
import WatchKit

class RegistrationController: WKInterfaceController
{
    @IBOutlet weak var userIdLabel: WKInterfaceLabel?
    @IBOutlet weak var channelIdLabel: WKInterfaceLabel?
    @IBOutlet weak var appKeyLabel: WKInterfaceLabel?
    var observer: NSObjectProtocol?
    
    override func willDisappear() {
        super.willDisappear()
        if let observer = observer
        {
            NotificationCenter.default.removeObserver(observer)
        }
    }
    
    override func willActivate() {
        super.willActivate()
        updateRegistrationLabels()
        observer = NotificationCenter.default.addObserver(forName: NSNotification.Name("MCERegisteredNotification"), object: nil, queue: OperationQueue.main, using: { (note) in
            self.updateRegistrationLabels()
        })
    }
    
    func updateRegistrationLabels()
    {
        guard let userIdLabel = userIdLabel else
        {
            return
        }
        guard let channelIdLabel = channelIdLabel else
        {
            return
        }
        guard let appKeyLabel = appKeyLabel else
        {
            return
        }
        
        if MCERegistrationDetails.sharedInstance().mceRegistered
        {
            userIdLabel.setText(MCERegistrationDetails.sharedInstance().userId)
            channelIdLabel.setText(MCERegistrationDetails.sharedInstance().channelId)
            appKeyLabel.setText(MCERegistrationDetails.sharedInstance().appKey)
        }        
    }
    
}

